packages/base/src/panelview/objecttree.tsx

```diff
@@ -237,6 +237,26 @@ class ObjectTreeReact extends React.Component<IProps, IStates> {
     this.setState(old => ({ ...old, lightTheme }));
   };
 
+  handleNodeClick = (objectId: string) => {
+    const object = this.getObjectFromName(objectId);
+
+    if (object && object.visible === true) {
+      const objPosition = object?.parameters?.Placement?.Position || {
+        x: 0,
+        y: 0,
+        z: 0
+      };
+
+      const event = new CustomEvent('nodeClick', {
```

Comment:
you need to add the `MainViewModel.id` to this event to be able to identify the correct 3D view

Reply:
Thanks @trungleduc, for that I'd need ho access `MainViewModel.id` in objectTree component to emit it correctly. I'll be on it and update :)

Reply:
Something like this in the `ControlPanelModel`:

```typescript
  get mainViewModel(): string | undefined {
    return (this._tracker.currentWidget as JupyterCadWidget | null)?.content
      .currentViewModel.id;
  }
```
Then in `ObjectTreeReact` you can use `this.props.cpModel.mainViewModel?.id` to get the current main view id

Reply:
Super @trungleduc, this helps so much 🚀